src/shared/util/objects.util.ts

```diff
@@ -15,7 +15,11 @@ export function deepMerge(obj1: object, obj2: object) {
 
   for (const key in obj2) {
     if (obj2.hasOwnProperty(key)) {
-      if (typeof obj2[key] === 'object' && typeof obj1[key] === 'object') {
+      if (
+        typeof obj2[key] === 'object' &&
+        typeof obj1[key] === 'object' &&
+        !Array.isArray(obj2[key])
+      ) {
```

Comment:
I know this whole method isn't a part of your PR.. 

But instead of reinventing the wheel here trying to maintain a custom deepmerge util,  maybe we should should replace this w/ something like https://www.npmjs.com/package/deepmerge-ts ? 🤔